src/thor/dao/release_dao.py

```diff
@@ -37,143 +39,141 @@ def session_scope():
-      #      rel.id, rel.version, rel.result
-      #  )
+        try:
+            release = session.query(Release).get(key)
+            assert release != None
+        except Exception as e:
+            log.info(
+                f"Attempted to retrieve key {key} from Releases, but could not locate. "
+            )
 
-      # log.info("### ##" + str(type(rel)))
-      log.info(f"retrieved release {rel} from the database...")
-      session.expunge_all()
-      return rel
+        # Note: check how many errors this throws if release breaks.
+        log.info(f"Retrieved release {release} from the database.")
+        session.expunge_all()
+        return release
 
 
-def readRelease(id):
-    with session_scope() as s:
-        # outstring = "The ID (" + str(id) + ") is not in the database. "
+def update_release(key, property, new_value):
+    """ Given the key of a release, the name of the property to be changed, 
+    and the intended new value of the property, change the value in the 
+    database to reflect the intended change. 
+    We assume that an entry with the key exists in the DB, that the property is a legit 
+    property name, and that the newValue is appropriate (same type). """
 
-        try:
-            rel = s.query(Release).get(id)
-            # print(rel, rel == None)
-            assert rel != None
-        except Exception as e:
-            pass
-            # print("The ID " + str(id) + " is not in the database. ")
-        # else:
-        #     outstring = "ID: '{}', Name: '{}', Version: '{}', Result: '{}'".format(
-        #         rel.id, rel.name, rel.version, rel.result
-        #     )
+    with session_scope() as session:
+        release = session.query(Release).get(key)
+        setattr(release, property, new_value)
+        log.info(f"Changed parameter {property} of entry {key} to {new_value}. ")
 
-        l.info(f"Retrieved release{rel} from the database. ")
-        s.expunge_all()
-        return rel
 
+def del_release(key):
+    """ Given the key of a particular Release, delete it from the table. 
+    If the key is not in the database, prints an error message. 
+    SUPERSEDED BY deleterelease, which should be more general. """
 
-def updateRelease(id, property, newValue):
     with session_scope() as session:
-      rel = session.query(Release).get(id)
-      setattr(rel, property, newValue)
-      l.info(f"Changed parameter {property} of {id} to {newValue}. ")
+        try:
+            if type(key) != int:
+                raise Exception(key)
+        except Exception as e:
+            print(str(e) + " is not an int. Check your types. ")
+            return
+
+        try:
+            session.delete(session.query(Release).get(key))
+        except Exception as e:
+            print("Cannot delete: " + str(key) + " is not in the database")
+            log.info(f"Entry {key} was deleted from Releases table. ")
+
 
+def delete_release(input):

```

Comment:
Don't have to address this right now, but I'd suggest having 2 different functions: `delete_release` and `delete_releases`.

`delete_releses` should make use to `delete_release` but only accept lists. This way you can move the if/elif logic upstream, rather than having the determining logic here, which will more likely hide problems when the system starts getting more complex.